[Fang Yiye] iP

yiyefyy opened the pull request:
# **Duke iP**

### A **_Bullet_** list
- todo
- deadline
- event
- - -
### A **_Numbered_** list
1. date
2. time
- - -
### A **_fenced_** code block
```ruby 
public void unmark(String input) {
        System.out.println("This task is marked as not done:");
        int j = Integer.parseInt(input.substring(7)) - 1;
        Task task = taskList.get(j);
        task.markAsNotDone();
        System.out.println(task);
    }
```
- - -
### A **_task_** list
- [X] eat 
- [ ] sleep
- - -
### An **_Emoji_**
👌
- - -
### Block Quote
> And now these three remain: faith, hope and love. But the greatest of these is love. 
- - -
### A hyper link 
[ip](https://github.com/yiyefyy/ip)
- - -
### Inline code
This function `public void greet() {
        System.out.println("Hello!, I'm Yiye.\nWhat can I do for you? ◠‿◠");
    }` greets you. 
- - -

yiyefyy committed:
no message

yiyefyy committed:
file is named as duke under ip

yiyefyy committed:
write method added

yiyefyy committed:
level-8 with command to search tasks using date
 commit message for your changes. Lines starting

yiyefyy committed:
gradle added

yiyefyy committed:
A-Gradle

yiyefyy committed:
organise into packages

yiyefyy committed:
updated organised files

yiyefyy committed:
OOP principle

yiyefyy committed:
modify file format

yiyefyy committed:
changed format

yiyefyy committed:
JUnit test added

yiyefyy committed:
Revert "JUnit test added"

This reverts commit d5faf4dead9ded67b708db9a8d76b6fc8df0b55a.

yiyefyy committed:
added text file

yiyefyy committed:
modified txt format

yiyefyy committed:
added jar file

yiyefyy committed:
export as jar

yiyefyy committed:
clean up

yiyefyy committed:
javaDoc for public classes and methods

yiyefyy committed:
Comply coding standard

yiyefyy committed:
Add method search name

yiyefyy committed:
no message

yiyefyy committed:
Merge branch 'master' into branch-A-JavaDoc

# Conflicts:
#	src/main/java/Duke/Duke.java
#	src/main/java/Duke/Parser.java
Fix conflicts

yiyefyy committed:
Merge branch 'branch-Level-9'

yiyefyy committed:
Resolve conflicts

# Conflicts:
#	src/main/java/Duke/Commands.java
#	src/main/java/Duke/Ui.java

yiyefyy committed:
Resolve conflicts

# Conflicts:
#	duke.txt
#	src/main/java/Deadline.java
#	src/main/java/Duke.java
#	src/main/java/Event.java
#	src/main/java/Task.java
#	src/main/java/Todo.java

aaronliwh commented on src/main/java/Duke/Commands.java:
```diff
@@ -0,0 +1,152 @@
+     * @param input read task index
+     */
+    public void markDone(String input) {
+        System.out.println("Nice! I have marked this task as done:");
```
Should this be handled by the command or the UI class or the command class ?

aaronliwh commented on src/main/java/Duke/Commands.java:
```diff
@@ -0,0 +1,152 @@
+ */
+public class Commands {
+    private final TaskList taskList;
+    private final Ui ui;
```
Would a single command class be any different from just adding the methods in Duke ?
Consider having an abstract Command class and having other commands extend from the Command class.

aaronliwh commented on src/main/java/Duke/Parser.java:
```diff
@@ -0,0 +1,56 @@
+     * @param input that is being read
+     * @throws DukeException when input is invalid
+     */
+    public void parse(String input) throws DukeException{
```
Should the error for invalid input (for example, unmark -100) be handled here ?
Also, consider using a switch statement instead

aaronliwh commented on src/main/java/Duke/Storage.java:
```diff
@@ -0,0 +1,76 @@
+package Duke;
+import java.io.*;
```
Try not to use * imports, instead import what you need explicitly

aaronliwh reviewed (commented):
I think your code is generally good, just a few changes to the code design that you may want to consider.

nqt230 commented on src/main/java/Duke/Duke.java:
```diff
@@ -0,0 +1,56 @@
+        ui = new Ui();
+        this.sc = new Scanner(System.in);
+        this.taskList = new TaskList();
+        filePath = "/Users/yiye/Desktop/cs2103Projects/ip/Data/duke.txt";
```
Would it better to declare the file path as a static constant in the Storage class? I think that this makes sense because Storage is supposed to handle all the file reading and writing operations, so the Duke class does not need to know the file path.

nqt230 commented on src/main/java/Duke/Duke.java:
```diff
@@ -0,0 +1,56 @@
+
+
+    /**
+     * run the program
```
I think the JavaDocs here could be more detailed in the parameter and exception descriptions, so for e.g.
`* @throws DukeException If the parser parses an invalid input.`

nqt230 reviewed (commented):
Looks good :+1:
Overall I think your code is easy to understand, with just a few concerns here and there regarding coding style and Javadocs, but nothing major.

yiyefyy committed:
level 10

yiyefyy committed:
no message

yiyefyy committed:
Enable Assertions

Project currently has no assert feature

Assertions enabled by setting it to true in build.gradle

Gradle was used to run the project

yiyefyy committed:
Move command lines to ui

There are some user interaction lines in methods of class commands.

This mades the code messy and unorganised.

lines such as System.out.println in methods markDone, unmark,
events, delete are removed. New methods are added in ui class
to replace them.

Each class should only contain methods that belong to the class
to avoid confusion.

yiyefyy committed:
Declare filepath in storage

filePath is declared in duke and passed as a parameter for
storage.

Duke class need not know the filePath hence it is unnecessary to declare it.

FilePath is now declared in storage as a final string constant.

FilePath is unchanged and only used in methods in storage hence only storage
needs to know the exact filePath.

yiyefyy committed:
Merge pull request #2 from yiyefyy/branch-A-Assertions

Enable Assertions

yiyefyy committed:
Edit JavaDoc

JavaDocs is short and lacks details

added details to javadoc such as the use of the params and exceptions thrown

this allows the method to be more clear and effective when debugging

yiyefyy committed:
Classify classes and use switch case

All classes are under one duke package and parser uses if-else statements which
has more than 3 indentations. One additional change is that  I have used an * for
an import statement.

I have changed the if-else to switch case in Parser class, classified the classed
according to they functions and listed out the import statements instead of using *

To:
- switch-case: make code more readable as it has less indentation
- classify: easier to look for classes to edit and debug
- state imports: avoid importing unused apis

yiyefyy committed:
Remove switch indentation

case has indentation

it was removed by changing the java code style in preference of intellij

to follow the java coding standard

yiyefyy committed:
Merge pull request #3 from yiyefyy/A-CodeQuality

A code quality

yiyefyy committed:
Fix javaDoc

header comment starts with add

does not follow coding standard

changed to adds

yiyefyy committed:
Added C tagging

all tasks has no tags

added tag command to tag task and print tags of a task.
Duke is now also able to search task of a certain tag.

by adding methods in commands class og guiUi class

yiyefyy committed:
added remove tag method

yiyefyy committed:
Change Gui format

yiyefyy committed:
more personality

yiyefyy committed:
Product image

yiyefyy committed:
Delete Screenshot 2022-09-18 at 7.31.40 PM.png

yiyefyy committed:
Add files via upload

yiyefyy committed:
Update README.md

yiyefyy committed:
Update README.md

yiyefyy committed:
changed back to if-else

error occurred when using switch case

yiyefyy committed:
Fixed some bugs

Some parts of the response printed is awkward

fixed the strings returned in classes commands, ui, and guiui

yiyefyy committed:
fix bugs

warning message has wrong date format

yiyefyy committed:
fixed some bugs

jar file could not run

yiyefyy committed:
rebuild jar file

yiyefyy committed:
rebuild project

yiyefyy committed:
Revert "rebuild project"

This reverts commit cac41c16c9da13bb489d19a7cd514c842cef9509.

yiyefyy committed:
changed launcher location

yiyefyy committed:
fixed bugs

yiyefyy committed:
new jar file

yiyefyy committed:
fixed filepath

yiyefyy committed:
fixed dialog box

yiyefyy committed:
save tag in txt file

yiyefyy committed:
fixed todo storage
